src/core/src/bootstrap/EnvLayer.py

```diff
@@ -162,9 +162,15 @@ def __str__(self):
 
     @staticmethod
     def __convert_process_output_to_ascii(output):
-        if sys.version_info.major == 2:
+        major_version = None
+        if hasattr(sys.version_info, 'major'):
+            major_version = sys.version_info.major
+        else:
+            major_version = sys.version_info[0]  # python 2.6 doesn't have attributes like 'major' within sys.version_info
```

Comment:
Okay to leave Container with duplicated code as it is its own thing. But extract this duplicated code into a separate function like __get_python_major_version() inside EnvLayer.